src/Acl.php

```diff
@@ -45,9 +60,30 @@ public function getRules(Model $model)
                 $modelClasses[] = $class;
             }
         } while (($class = get_parent_class($class)) !== false);
-        $res = $user->ref('AccessRules')->addCondition('model', 'in', $modelClasses);
 
-        return $res; // @phpstan-ignore-line
+        // Internally disable ACL for a moment and limit to only required fields to avoid recursion
+        $this->disabled = true;
+        $rules = $user->ref('AccessRules')
```

Comment:
try {} finally { $this->disabled = false; }

also rename `$disabled` to something like `skipApplyRestrictionsForRulesExport` - then the phpdoc comment can be removed, comments are usually bad than helpful, code is always better :)

Reply:
I love comments :D :D :D

Reply:
comments needs to carry value :D :D :D